intentgen/src/main/java/com/chatbot/IntentGenerator.java

```diff
@@ -0,0 +1,184 @@
+package com.chatbot;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import com.chatbot.protobuf.UseCaseOuterClass.UseCase;
+import com.google.cloud.dialogflow.v2.Intent;
+import com.google.cloud.dialogflow.v2.Context;
+import com.google.cloud.dialogflow.v2.Intent.TrainingPhrase;
+import com.google.cloud.dialogflow.v2.Intent.TrainingPhrase.Part;
+import com.google.cloud.dialogflow.v2.IntentsClient;
+import com.google.cloud.dialogflow.v2.ProjectAgentName;
+import com.google.protobuf.TextFormat;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+
+public class IntentGenerator {
+  static final List<String> YES_TRAINING_PHRASES = new ArrayList<String>(Arrays.asList("yes",
+      "okay I will", "why not", "yes that's alright", "yes I do", "exactly", "of course",
+      "yep that's ok", "okay", "ok", "sure"));
+  static final List<String> NO_TRAINING_PHRASES = new ArrayList<String>(Arrays.asList("No",
+      "thanks but no", "no way", "no no don't", "na", "no it isn't", "don't", "nah I'm good",
+      "no I cannot", "I can't"));
+  static final int ENTITY_TYPE_INDEX = 0;
+  static final int ALIAS_INDEX = 1;
+  static final int VALUE_INDEX = 2;
+  private static Map<String, String> intentDisplayNameToName;
+
+  public static void main(final String[] args) throws ParseException {
+    Options options = new Options();
+    final Option useCaseFile = Option.builder("f").hasArg()
+        .desc("prototxt file describing the use case" ).build();
+    options.addOption(useCaseFile);
+    final CommandLineParser parser = new DefaultParser();
+    final CommandLine cmd = parser.parse(options, args);
+    final String fileName = cmd.getOptionValue("f");
+    if(fileName.isEmpty()) {
+      throw new ParseException("No file name provided");
```

Comment:
InvalidArgumentException if this is not available in

Reply:
updates